src/main/java/com/foo/gosucatcher/domain/expert/presentation/ExpertController.java

```diff
@@ -49,17 +64,66 @@ public ResponseEntity<ExpertsResponse> findAll() {
 		return ResponseEntity.ok(experts);
 	}
 
-	@PatchMapping("/{id}")
-	public ResponseEntity<Long> update(@PathVariable Long id, @Validated @RequestBody ExpertUpdateRequest request) {
-		Long expertId = expertService.update(id, request);
+	@PatchMapping("/{expertId}")
+	public ResponseEntity<Long> update(@PathVariable Long expertId,
+		@Validated @RequestBody ExpertUpdateRequest request) {
+		Long updatedExpertId = expertService.update(expertId, request);
+
+		return ResponseEntity.ok(updatedExpertId);
+	}
+
+	@DeleteMapping("/{expertId}")
+	public ResponseEntity<Void> delete(@PathVariable Long expertId) {
+		expertService.delete(expertId);
+
+		return ResponseEntity.ok(null);
+	}
+
+	@PostMapping("/{expertId}/images")
+	public ResponseEntity<ImageUploadResponse> uploadImage(@PathVariable Long expertId, MultipartFile file) throws
+		IllegalStateException,
+		IOException {
+		ImageUploadRequest request = new ImageUploadRequest(expertId, file);
+		String uploadedFilename = imageService.store(request);
 
-		return ResponseEntity.ok(expertId);
+		ImageUploadResponse response = new ImageUploadResponse(expertId, uploadedFilename);
+		return ResponseEntity.status(HttpStatus.CREATED).body(response);
 	}
 
-	@DeleteMapping("/{id}")
-	public ResponseEntity<Void> delete(@PathVariable Long id) {
-		expertService.delete(id);
+	@GetMapping("/{expertId}/images/{filename}")
+	public ResponseEntity<Resource> getImage(@PathVariable Long expertId,
+		@PathVariable String filename) {
+
+		Resource file = imageService.loadAsResource(expertId, filename);
+		return ResponseEntity.ok().header(HttpHeaders.CONTENT_DISPOSITION,
+			"attachment; filename=\"" + file.getFilename() + "\"").body(file);
```

Comment:
이런식으로 변수로 뺴는 것 어떤가요?

```suggestion
		String responseHeader = "attachment; filename=\"" + file.getFilename() + "\"";
		return ResponseEntity.ok().header(HttpHeaders.CONTENT_DISPOSITION, responseHeader).body(file);
```

Reply:
말씀주신 방향이 훨씬 깔끔한것같습니다! 👍👍 감사합니다!!